hal/modem_api.h

```diff
@@ -0,0 +1,88 @@
+/* mbed Microcontroller Library
```

Comment:
device_xx is a chip resource, not a board resource. If you look at every other similar example (sd, esp, nor, accel, gyro, etc. you'll find that the 'component' approach is what sets precedent and should be followed.